cart/application/eventPublishing.go

```diff
@@ -33,11 +33,16 @@ type (
 		QtyAfter               int
 	}
 
+	// PaymentSelectionHasBeenResetEvent defines event properties
+	PaymentSelectionHasBeenResetEvent struct {
+	}
+
 	//EventPublisher - technology free interface to publish events that might be interesting for outside (Publish)
 	EventPublisher interface {
 		PublishAddToCartEvent(ctx context.Context, marketPlaceCode string, variantMarketPlaceCode string, qty int)
 		PublishChangedQtyInCartEvent(ctx context.Context, item *cartDomain.Item, qtyBefore int, qtyAfter int, cartID string)
 		PublishOrderPlacedEvent(ctx context.Context, cart *cartDomain.Cart, placedOrderInfos cartDomain.PlacedOrderInfos)
+		PublishPaymentSelectionHasBeenResetEvent(ctx context.Context)
```

Comment:
I think passing the cart reference makes sense and gives the event the correct context?

Reply:
I'll added the cart reference, please have a look